digiwf-engine/digiwf-engine-service/src/main/java/de/muenchen/oss/digiwf/process/definition/domain/service/ServiceInstanceInitializationService.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c): it@M - Dienstleister für Informations- und Telekommunikationstechnik der Landeshauptstadt München, 2020
+ */
+
+package de.muenchen.oss.digiwf.process.definition.domain.service;
+
+import de.muenchen.oss.digiwf.process.config.domain.model.ProcessConfig;
+import de.muenchen.oss.digiwf.process.config.domain.service.ProcessConfigService;
+import de.muenchen.oss.digiwf.process.instance.domain.model.ServiceInstance;
+import de.muenchen.oss.digiwf.process.instance.domain.service.ServiceInstanceService;
+import de.muenchen.oss.digiwf.process.instance.process.ProcessConstants;
+import de.muenchen.oss.digiwf.process.instance.process.properties.S3Properties;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.camunda.bpm.engine.RepositoryService;
+import org.camunda.bpm.engine.RuntimeService;
+import org.camunda.bpm.engine.repository.ProcessDefinition;
+import org.camunda.bpm.engine.variable.Variables;
+import org.springframework.stereotype.Service;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static de.muenchen.oss.digiwf.process.instance.process.ProcessConstants.PROCESS_S3_ASYNC_CONFIG;
+import static de.muenchen.oss.digiwf.process.instance.process.ProcessConstants.PROCESS_S3_SYNC_CONFIG;
+
+/**
+ * Service to initialize service instances.
+ *
+ * @author externer.dl.horn
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ServiceInstanceInitializationService {
+
+    private final RepositoryService repositoryService;
+    private final RuntimeService runtimeService;
+
+    private final ServiceInstanceService serviceInstanceService;
+    private final ProcessConfigService processConfigService;
+    private final S3Properties s3Properties;
+
+
+    public void initializeInstance(final String processInstanceId, final String key, final Map<String, Object> currentVariables) {
+
+        final String processDefinitionName = this.getProcessDefinitionName(key);
+
+        //create service instance
+        final ServiceInstance serviceInstance = this.serviceInstanceService.creatServiceInstance(processDefinitionName, key);
+        this.serviceInstanceService.updateInstanceId(serviceInstance.getId(), processInstanceId);
```

Comment:
Kann man sicherstellen, dass der Listener vor dem ersten Job abläuft (egal ob Synchron oder Asynchron)

Eventuell haben wir hier ein Problem mit der Reihenfolge der Abarbeitung

Siehe: https://github.com/it-at-m/digiwf-core/pull/1399

Reply:
setStatus funktioniert noch. Habe 2 Testprozesse hinzugefügt, die das bestätigen 👍